src/approaches/interactive_learning_approach.py

```diff
@@ -255,6 +258,75 @@ def _create_glib_interaction_strategy(
         termination_function = task.goal_holds
         return act_policy, termination_function
 
+    def _create_greedy_lookahead_interaction_strategy(
+        self, train_task_idx: int
+    ) -> Tuple[Callable[[State], Action], Callable[[State], bool]]:
+        """Sample a certain number of max-length trajectories and pick the one
+        that has the highest cumulative score."""
+        init = self._train_tasks[train_task_idx].init
+        # Sample trajectories by sampling random sequences of NSRTs.
+        best_score = -np.inf
+        best_options = []
+        for _ in range(CFG.interactive_max_num_trajectories):
+            state = init.copy()
+            options = []
+            trajectory_length = 0
+            total_score = 0.0
+            while trajectory_length < CFG.interactive_max_trajectory_length:
+                # Sample an NSRT that has preconditions satisfied in the
+                # current state.
+                ground_nsrt = self._sample_applicable_ground_nsrt(state)
```

Comment:
inside this function, you're re-computing all the ground nsrts, but that only has to be done once at the start of this function (using the objects in `init`). so, we should just make _sample_applicable_ground_nsrt take in all the ground  nsrts.